kodev

```diff
@@ -399,6 +399,36 @@ TARGET:
 }
 
 
+function kodev-copy {
+    LOG_HELP_MSG="
+usage: copy <Root folder of KOReader on target device>
+
+Example: ./kodev copy /media/hzj_jie/HZJ-VOYAGE/koreader
+    "
+
+    [ $# -lt 1 ] && { echo "${LOG_HELP_MSG}"; exit 1; }
+
+    case $1 in
+        -h | --help)
+            echo "${LOG_HELP_MSG}" ;;
+        *)
+            for i in $(git diff upstream/master --stat |
+                       grep '|' |
+                       sed 's/ /\t/g' |
+                       cut -f 2 |
+                       grep -v "^kodev$")
+            do
+                cp $i $1/$i
```

Comment:
We can still just rsync frontend and plugins directories as the initial implementation. But if this is the case, we should put a notice in the help message to let users know it only handles lua sources from koreader. When we added support to also include everything in the base, we can add an extra flag like `--frontend` to restrict the push to only frontend lua files.

rsync can also take care of purging unused files on the device for you. Plus it gives you a real diff between files on the device v.s. files on your development machine. Getting the diff from git can lead to issues when what's installed on the device is not the same version as `upstream/master`.

Overall, I think this is a very useful feature and I would totally use it myself ;)